fxa-oauth-server/lib/routes/token.js

```diff
@@ -375,43 +360,30 @@ function generateIdToken(options, access) {
   return ID_TOKEN_KEY.sign(claims);
 }
 
-function generateTokens (options) {
-  // we always are generating an access token here
-  // but depending on options, we may also be generating a refresh_token
-  return db.generateAccessToken(options)
-    .then((access) => {
-      const promises = {};
-      if (options.offline) {
-        promises.refresh = db.generateRefreshToken(options);
-      }
-      if (options.idToken) {
-        promises.idToken = generateIdToken(options, access);
-      }
-
-      return P.props(promises).then(function (result) {
-        const refresh = result.refresh;
-        const idToken = result.idToken;
-
-        const json = {
-          access_token: access.token.toString('hex'),
-          token_type: access.type,
-          scope: access.scope.toString()
-        };
-        if (options.authAt) {
-          json.auth_at = options.authAt;
-        }
-        json.expires_in = options.ttl;
-        if (refresh) {
-          json.refresh_token = refresh.token.toString('hex');
-        }
-        if (idToken) {
-          json.id_token = idToken;
-        }
-        if (options.keysJwe) {
-          json.keys_jwe = options.keysJwe;
-        }
-        return json;
-      });
-    });
+async function generateTokens(options) {
+  // We always generate an access_token.
+  const access = await db.generateAccessToken(options);
+  const result = {
+    access_token: access.token.toString('hex'),
+    token_type: access.type,
+    scope: access.scope.toString()
+  };
+  result.expires_in = options.ttl;
+  if (options.authAt) {
+    result.auth_at = options.authAt;
+  }
+  if (options.keysJwe) {
+    result.keys_jwe = options.keysJwe;
+  }
+  // Maybe also generate a refreshToken?
+  if (options.offline) {
```

Comment:
If a refreshToken was used to generate this token, will a new refreshToken be generated? From the [db schema](https://github.com/mozilla/fxa-auth-server/blob/40b65eeb05a11437ae32cbf4b2ed06e30d71f261/fxa-oauth-server/lib/db/mysql/schema.sql#L74) it doesn't look like that's possible. Just want to make sure.

Reply:
It shouldn't be possible, but we should probably make that more explicit.